monai/transforms/spatial/functional.py

```diff
@@ -610,3 +611,68 @@ def affine_func(
         out = _maybe_new_metatensor(img, dtype=torch.float32, device=resampler.device)
     out = out.copy_meta_from(meta_info) if isinstance(out, MetaTensor) else out
     return out if image_only else (out, affine)
+
+
+def convert_box_to_points(bbox, mode):
+    """
+    Convert bounding box to points.
+
+    Args:
+        mode: The mode specifying how to interpret the bounding box.
+        bbox: Bounding box in the form of [x1, y1, x2, y2] for 2D or [x1, y1, z1, x2, y2, z2] for 3D.
+            Return shape will be (N, 4) for 2D or (N, 6) for 3D.
+
+    Returns:
+        sequence of points representing the corners of the bounding box.
+    """
+
+    mode = get_boxmode(mode)
+
+    points_list = []
+    for _num in range(bbox.shape[0]):
+        corners = mode.boxes_to_corners(bbox[_num : _num + 1])
+        if len(corners) == 4:
+            points_list.append(
+                concatenate(
+                    [
+                        concatenate([corners[0], corners[1]], axis=1),
+                        concatenate([corners[2], corners[1]], axis=1),
+                        concatenate([corners[2], corners[3]], axis=1),
+                        concatenate([corners[0], corners[3]], axis=1),
+                    ],
+                    axis=0,
+                )
+            )
+        else:
+            points_list.append(
+                concatenate(
+                    [
+                        concatenate([corners[0], corners[1], corners[2]], axis=1),
+                        concatenate([corners[3], corners[1], corners[2]], axis=1),
+                        concatenate([corners[3], corners[4], corners[2]], axis=1),
+                        concatenate([corners[0], corners[4], corners[2]], axis=1),
+                        concatenate([corners[0], corners[1], corners[5]], axis=1),
+                        concatenate([corners[3], corners[1], corners[5]], axis=1),
+                        concatenate([corners[3], corners[4], corners[5]], axis=1),
+                        concatenate([corners[0], corners[4], corners[5]], axis=1),
+                    ],
+                    axis=0,
+                )
+            )
+
+    return stack(points_list, dim=0)
+
+
+def convert_points_to_box(points):
+    """
+    Convert points to bounding box.
+
+    Args:
+        points: Points representing the corners of the bounding box. Shape (N, 8, 3) or (N, 4, 2).
+    """
```

Comment:
```suggestion
    """
    Converts points to an axis-aligned bounding box.

    Args:
        points: Points representing the corners of the bounding box. Shape (N, 8, 3) for the 8 corners of
        a 3D cuboid or (N, 4, 2) for the 4 corners of a 2D rectangle.
    """
```

This could actually work for any arbitrary number of 2D or 3D points provided at least two non-identical points are given.

Reply:
The clarifications I've mentioned here and in the previous comment should perhaps be reflected in the docstrings of the added transforms as well.

Reply:
Thanks! Updated the docstring.